File: python-leetcode/array/73.矩阵置零.py
from typing import List


class Solution:
    """
    Do not return anything, modify matrix in-place instead.
    """

    def setZeroes(self, matrix: List[List[int]]) -> None:
        # 第一次遍历，发现 matrix[i][j] === 0 ,则设置 rows[i] = true,col[j] = true
        # 第二次遍历，发现存在rows[i] = true ｜ col[j] = true，则置为 0
        rows = [False] * len(matrix)
        cols = [False] * len(matrix[0])

        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if matrix[i][j] == 0:
                    rows[i] = True
                    cols[j] = True

        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if rows[i] or cols[j]:
                    matrix[i][j] = 0

File: go-leetcode/array/73.矩阵置零.go
package leetcode

// Do not return anything, modify matrix in-place instead.
func setZeroes(matrix [][]int) {
	// 第一次遍历，发现 matrix[i][j] === 0 ,则设置 rows[i] = true,col[j] = true
	// 第二次遍历，发现存在rows[i] = true ｜ col[j] = true，则置为 0
	rows := make([]bool, len(matrix))
	cols := make([]bool, len(matrix[0]))

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[i]); j++ {
			if matrix[i][j] == 0 {
				rows[i] = true
				cols[j] = true
			}
		}
	}

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[i]); j++ {
			if rows[i] || cols[j] {
				matrix[i][j] = 0
			}
		}
	}
}
